src/Oxid/Core/MakairaTrackingDataGenerator.php

```diff
@@ -0,0 +1,186 @@
+
+/**
+ * This file is part of a marmalade GmbH project
+ * It is not Open Source and may not be redistributed.
+ * For contact information please visit http://www.marmalade.de
+ * @version    0.1
+ * @author     Sunny <dt@marmalade.de>
+ * @link       http://www.marmalade.de
+ */
+
+/**
+ * This class generates tracking data for certain pages in OXID.
+ * To add more pages create a new "generateFor<ControllerClass>" method.
+ * For example <ControllerClass> can be "Content" for OXIDs CMS content controller.
+ */
+class MakairaTrackingDataGenerator
+{
+    public const TRACKER_URL = 'https://piwik.makaira.io/';
+
+    private static mixed $odoScopeTracking = false;
+
+    /**
+     * Generates the tracking data array to send to piwik.
+     * @param string $oxidControllerClass OXIDs controller class to generate the tracking data for.
+     * @return array
+     * @throws ArticleInputException
+     * @throws NoArticleException
+     */
+    public function generate(string $oxidControllerClass): array
+    {
+        $siteId = Registry::getConfig()->getShopConfVar(
+            'makaira_tracking_page_id',
+            null,
+            'module:makaira_oxid-connect-essential'
+        );
+
+        if (empty($siteId)) {
+            return [];
+        }
+
+        $childTrackingData = null;
+        $normalizedClass = $this->normalize($oxidControllerClass);
+        $methodName = "generateFor{$normalizedClass}";
+
+        if (is_callable([$this, $methodName]) && method_exists($this, $methodName)) {
+            $childTrackingData = $this->{$methodName}();
+        }
+
+        if (Registry::getSession()->getBasket()->isNewItemAdded()) {
+            $childTrackingData = $this->generateForBasket();
+        }
+
+        if (null === $childTrackingData) {
+            $childTrackingData = [['trackPageView']];
+        }
+
+        $trackingData = [
+            $childTrackingData,
+            [
+                ['enableLinkTracking'],
+                ['setTrackerUrl', "{$this->getTrackerUrl()}/piwik.php"],
+                ['setSiteId', $siteId],
+            ],
+            $this->getCustomTrackingData()
+        ];
+
+        if (self::$odoScopeTracking) {
+            $trackingData[] = [
+                ['trackEvent', 'odoscope', self::$odoScopeTracking['group'], self::$odoScopeTracking['data']]
+            ];
+        }
+
+        // Send tracking to piwik if mak_experiments cookie is set
+        if (!empty($_COOKIE['mak_experiments'])) {
+            $experiments = json_decode($_COOKIE['mak_experiments'], true);
+            if (is_array($experiments)) {
+                foreach ($experiments as $experiment) {
+                    $trackingData[] = [
+                        [
+                            'trackEvent',
+                            'abtesting',
+                            $experiment['experiment'],
+                            $experiment['variation']
+                        ]
+                    ];
+                }
```

Comment:
Alternative for the `foreach`:
```suggestion
                $trackingData[] = array_map(
                    static fn ($experiment) => [
                        'trackEvent',
                        'abtesting',
                        $experiment['experiment'],
                        $experiment['variation']
                    ],
                    $experiments
                );
```